### PR TITLE
Feature/application/hello uefi

### PR DESCRIPTION
**Pull request by Flickdm**
## Description

This creates the simplest possible hello world for a UEFI application

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

It was ran on QemuQ35

## Integration Instructions

N/A

### CODE REVIEW

File: MdeModulePkg/Application/HelloUefi/HelloUefi.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+EFI_STATUS
+EFIAPI
+UefiMain (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_STATUS  Status;
+
+  Status = SystemTable->ConOut->OutputString (SystemTable->ConOut, L"Hello Uefi!\r\n");
```
Comment (apop5) [inline MdeModulePkg/Application/HelloUefi/HelloUefi.c:37]: [You need to check that ConOut is valid before you use it. ](https://github.com/microsoft/mu_basecore/blob/release/202302/MdeModulePkg/Core/Dxe/DxeMain/DxeMain.c#L136C4-L136C4)

You can check that ConsoleOutHandle is not NULL

File: MdeModulePkg/Application/HelloUefi/HelloUefi.inf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  HelloUefi.c
+  HelloUefi.ver
+
+[LibraryClasses]
+  UefiApplicationEntryPoint
+  DebugLib
+
+[Packages]
+  MdePkg/MdePkg.dec
+  MdeModulePkg/MdeModulePkg.dec
```
Comment (os-d) [inline MdeModulePkg/Application/HelloUefi/HelloUefi.inf]: nit newline at EOF

File: MdeModulePkg/Application/HelloUefi/HelloUefi.ver [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "FileVersion" : "1.0.0.0",
+    "OriginalFilename" : "HelloUefi.efi",
+    "CompanyName" : "Example Company Name"
+}
```
Comment (os-d) [inline MdeModulePkg/Application/HelloUefi/HelloUefi.ver]: nit newline at EOF

File: MdeModulePkg/Application/HelloUefi/HelloUefi.inf [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  MODULE_UNI_FILE                = HelloUefi.uni
+  FILE_GUID                      = 1C50FE8C-8607-4CB0-9050-48F6EE46BB82
+  MODULE_TYPE                    = UEFI_APPLICATION
+  VERSION_STRING                 = 1.0
+  ENTRY_POINT                    = UefiMain
+
+#
+# The following information is for reference only and not required by the build tools.
+#
+#  VALID_ARCHITECTURES           = IA32 X64 EBC
```
Comment (os-d) [inline MdeModulePkg/Application/HelloUefi/HelloUefi.inf]: small nit: can drop EBC, add AARCH64

**Review by os-d — COMMENTED** [review]
LGTM, I would just recommend putting this is in mu_oem_sample or mu_tiano_platforms instead of mu_basecore.

**Review by os-d — CHANGES_REQUESTED** [review]
LGTM, I would just recommend putting this is in mu_oem_sample or mu_tiano_platforms instead of mu_basecore.

**Pull request closed without merging**